tinkoff/invest/strategy/signal_executor.py

```diff
@@ -0,0 +1,75 @@
+import abc
+from abc import ABC
+from functools import singledispatchmethod
+
+from tinkoff.invest.async_services import AsyncServices
+from tinkoff.invest.strategy.errors import UnknownSignal
+from tinkoff.invest.strategy.signal import (
+    CloseLongMarketOrder,
+    CloseShortMarketOrder,
+    OpenLongMarketOrder,
+    OpenShortMarketOrder,
+    Signal,
+)
+from tinkoff.invest.strategy.strategy import MovingAverageStrategyState
+
+
+class ISignalExecutor(abc.ABC):
```

Comment:
сделай это протоколом

```suggestion
class ISignalExecutor(typing.Protocol):
```

Нам по сути важен интерфейс, а не абстрактный класс